### PR TITLE
Add insecure repository support

### PR DESCRIPTION
**Pull request by fahhem**
Local registries, such as microk8s, don't have SSL setup.

This PR just adds the ability to specify to push to an insecure repository. I'm not sure if that's the intended design pattern for this, so I'm open to suggestions!

### CODE REVIEW

**Comment by fahhem** [comment]
This might also address #1245

**Review by smukherj1 — COMMENTED** [review]
Thanks for the PR! Do you mind adding a test [here](https://github.com/bazelbuild/rules_docker/blob/master/testing/e2e/pusher.sh) that exercises the new insecure_repository attribute?

**Comment by jake-arkinstall** [comment]
Any update on this?

**Comment by asv** [comment]
@fahhem will you finish this PR or can I post mine? For our production, this is a very important feature in rules_docker.

**Comment by fahhem** [comment]
I don't have a publicly-available insecure repository available to add a test against. If you have a better PR, go ahead and post it, I'm not attached to this PR in any way. I was simply sharing what I had to do to get a on-machine docker repository to work.

**Comment by fahhem** [comment]
I've rebased onto master. I don't know the test infra here, so I added a "test" that is probably wrong. :shrug: if anyone else has a better test or at least a suggestion on how to write one, I'm happy to swap mine out for it.

**Comment by cfife-btig** [comment]
We are very interested in having this merged in. We are also dependent on a registry which is insecure and are having to write lots of workarounds without this change.

**Comment by fahhem** [comment]
Rebased again. This time the tests are failing for the same reason the master tests are failing (something about an unsupported version of ubuntu?  `ubuntu2104` vs the mentioned `ubuntu1804`)

**Comment by fahhem** [comment]
@uhthomas any chance this can be reviewed? It's a pretty small PR :)

**Review by linzhp — APPROVED** [review]
LGTM

**Comment by fahhem** [comment]
@linzhp any idea on what I should do about the required-but-failing CI step? Or will you be able to merge it anyway?

**Comment by fahhem** [comment]
Oh, and thank you!

**Comment by linzhp** [comment]
It doesn't allow me to merge without CI passing

File: container/go/cmd/pusher/pusher.go [hunk 1]
```diff
@@ -126,7 +127,12 @@ func main() {
 		log.Printf("Failed to digest image: %v", err)
 	}
 
-	if err := push(stamped, img); err != nil {
+	var options = []name.Option{};
```
Comment (uhthomas) [inline container/go/cmd/pusher/pusher.go:130]: This needs to be run through gofmt, Go code does not typically have semi colons.

Also, it's usually preferred to write this as:

```go
var opts []name.Option
```

File: container/go/cmd/pusher/pusher.go [hunk 2]
```diff
@@ -126,7 +127,12 @@ func main() {
...
+	if *insecureRepository {
+		options = append(options, name.Insecure);
```
Comment (uhthomas) [inline container/go/cmd/pusher/pusher.go:132]: gofmt

**Review by uhthomas — CHANGES_REQUESTED** [review]

**Comment by cfife-btig** [comment]
I'm just happy to see some progress is on this MR :')

**Comment by fahhem** [comment]
Same here :D Thanks @uhthomas, I ran gofmt and it had a couple more changes than around my code, but it was all whitespace stuff.

**Comment by fahhem** [comment]
sorry for the noise, github's reviewer UI looks... strange. Anyone consider reviewable.io?

**Comment by fahhem** [comment]
@uhthomas sorry to bother!

**Comment by cfife-btig** [comment]
I don't mean to keep pecking either. But my team is also really interested in getting this merged in as well. I know that the CI is still being blocked but it sorta sounds like it just might be some sort of noise and not anything legitimate. Although of course I also have no visibility.

**Review by uhthomas — APPROVED** [review]

**Comment by uhthomas** [comment]
Looks okay, we'll need to understand why CI isn't happy and then we can merge.

**Comment by fahhem** [comment]
All PR's are blocked until CI is fixed? Is anyone at bazel working on fixing the CI?

It's a lot easier to fix buildkite stuff when you have access to the servers, but I can take a stab at it still.

**Comment by linzhp** [comment]
Sorry for the delay. I just fixed CI (#2178)

**Comment by fahhem** [comment]
Thank you @linzhp I fixed the tests too, so this PR should be ready to go!

**Comment by cfife-btig** [comment]
🎉🎉🎉

**Comment by yc185050** [comment]
Any idea when will this be released? Waiting on this fix.

**Comment by fahhem** [comment]
@yc185050 I recommend opening a new issue to request a release. Personally, I'm using master at that merged commit as my company's release while we wait for an official release

**Comment by matthewei** [comment]
Hi , how can we use this feature?

**Pull request merged**